crypto/x509/x_x509.c

```diff
@@ -97,6 +97,17 @@ static int x509_cb(int operation, ASN1_VALUE **pval, const ASN1_ITEM *it,
         ASN1_OCTET_STRING_free(ret->distinguishing_id);
         break;
 
+    case ASN1_OP_DUP_POST:
+        {
+            X509 *old = exarg;
```

Comment:
@DDvO : are you sure that every `exarg` you get here is really a `X509` type one ?
I have a doubt, may be unfunded.
I may debug a bit of test code on a 32 bit windows environment, if needed ?

Reply:
Interesting thought -
I do not see how `x509_cb` could be called in any other context than for pointers of type `X509 *`, but who knows.
So if you can debug this on such an environment, would be great in particular to obtain a backtrace that leads to the situation currently excluded by the workaround
```
#ifdef OPENSSL_SYS_WINDOWS
    /*
     * Workaround for following issue with Windows VS 15 shared x86 builds:
     * 74080000:error::asn1 encoding routines:asn1_item_embed_d2i:illegal options on item template:..\crypto\asn1\tasn_dec.c:178:Type=X509_NAME_INTERNAL
     * 74080000:error::asn1 encoding routines:ASN1_item_dup:aux error:..\crypto\asn1\a_dup.c:83:Type=X509_NAME
    */
    if (asn1_cb != NULL && strcmp(it->sname, "X509_NAME") == 0)
        asn1_cb = NULL;
#endif
```
So please replace the `#ifdef OPENSSL_SYS_WINDOWS` e.g. by `#if 1` and set breakpoint on `asn1_cb = NULL;`
and then let's analyze how execution arrived there.